tests/conftest.py

```diff
@@ -948,6 +948,67 @@ def projects_dir() -> str:
     yield os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")
 
 
+@pytest.fixture(scope="function")
+def project_builder(tmpdir: py.path.local) -> Callable[[str, List[str], List[str]], Project]:
```

Comment:
Isn't this very similar to what the snippetcompiler does?

Perhaps the one can use the other?